lib/bloc/pos_catalog/bloc.dart

```diff
@@ -103,6 +108,30 @@ class PosCatalogBloc with AsyncActionsHandler {
     _loadItems(filter: action.filter);
   }
 
+  _exportItems(ExportItems action) async {
+    List<Item> itemList = await _repository.fetchItems();
+    if (itemList.length != 0) {
+      action.resolve(await PosCsvUtils(itemList: itemList).export());
+    } else {
+      throw Exception("EMPTY_LIST");
+    }
+  }
+
+  _importItems(ImportItems action) async {
+    List<Item> itemList =
+        await PosCsvUtils().retrieveItemListFromCSV(action.importFile);
+    // We should try to restore old db if import fails.
+    // var backupDB = await _repository.fetchItems();
+    await resetDB();
```

Comment:
Resetting the db and importing the new items should be executed in one db transaction otherwise we may end with empty database.

Reply:
[Done](https://github.com/breez/breezmobile/pull/357/commits/875ac92106e860ae72da9b32100f4797ffadfdbc)